src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -122,7 +122,13 @@ void NrfWiFiDriver::OnNetworkStatusChanged(Status status)
 
     if (mpNetworkStatusChangeCallback)
     {
-        mpNetworkStatusChangeCallback->OnNetworkingStatusChange(status, NullOptional, NullOptional);
+        WiFiManager::WiFiInfo wifiInfo;
+
+        if (CHIP_NO_ERROR == WiFiManager::Instance().GetWiFiInfo(wifiInfo))
```

Comment:
Should we fallback to NullOptional in case the GetWiFiInfo() fails for whatever reason?

Reply:
This might be okay, however I'm not sure. I see that in all 15 platforms the correct NetworkID is used, or `OnNetworkingStatusChange` is skipped.

Reply:
But don't we support only one network anyway? If the last connection failed then we can't interact with the device anyway, right?

Reply:
Yeah, but the connection may also fail during commissioning.

Reply:
discussed f2f, will be fixed later as it's not vital for now.